atomic_reactor/plugins/build_orchestrate_build.py

```diff
@@ -389,26 +429,42 @@ def do_worker_build(self, release, cluster_info, koji_upload_dir, task_id):
                 except OsbsException:
                     pass
 
+    def select_and_start_cluster(self, platform):
+        ''' Choose a cluster and start a build on it '''
+
+        config = get_config(self.workflow)
+        clusters = config.get_enabled_clusters_for_platform(platform)
+
+        if not clusters:
+            raise UnknownPlatformException('No clusters found for platform {}!'
+                                           .format(platform))
+
+        possible_clusters = None
+        retry_info = self.ClusterRetryContexts(clusters, self.max_cluster_fails)
+
+        while True:
+            if not clusters:
+                raise RuntimeError('Could not find appropriate cluster for worker build.')
+            possible_clusters = self.get_clusters(clusters, platform, retry_info)
+            for cluster in possible_clusters:
+                try:
+                    self.do_worker_build(self.release, cluster, self.koji_upload_dir,
+                                         self.fs_task_id)
+                    return
+                except BuildCanceledException:
+                    raise
+                except Exception:
```

Comment:
Would it be safer to only retry on `OsbsException`?

Reply:
Tim had me change it to Exception

Reply:
here: https://github.com/projectatomic/atomic-reactor/pull/841#discussion_r140527626

I'd be fine with changing this to OsbsException instead (and removing the BuildCanceledException clause) if everyone's happy with that and agrees that do_worker_build() shouldn't ever raise anything else.